src/ophyd_async/epics/_backend/common.py

```diff
@@ -10,9 +12,11 @@ def get_supported_values(
     if not datatype:
         return {x: x or "_" for x in pv_choices}
 
-    if not issubclass(datatype, str):
+    if issubclass(datatype, RuntimeEnum):
+        pv_choices = datatype.choices
```

Comment:
Here we need to check that `pv_choices` are a strict subset of `datatype.choices`, then we can discard datatype, we don't need to make `pv_choices` from it

Reply:
I'm a little confused by what you're saying here...

Surely we check `datatype.choices ⊆ pv_choices`. There could be `pv_choices = {"A", "B", ..., "Z"}` on the epics record, and the user might have wrote code to deal with `datatype = RuntimeEnum["A", "B"]`.

If `pv_choices ⊆ datatype.choices` then we'd be allowing `RuntimeEnum["A", ..., "Z"]` for a record with only `pv_choices = {"A"}`.

Reply:
Sorry, I got subset and superset mixed up!